### ISSUE TITLE
More precise create_time (and boot_time, etc.) on Windows

### ISSUE THREAD

**Issue by gaborcsardi** [giampaolo/psutil#1302]
I think sg. like this should work:

```c
long long   ll, secs, nsecs;
FILETIME    ft;

...

ll = ((LONGLONG) ft.dwHighDateTime) << 32;
ll += ft.dwLowDateTime - 116444736000000000LL;
secs = ll / 10000000;
nsecs = ll % 10000000;
return (double) secs + ((double) nsecs) / 10000000;
```

**Comment by gaborcsardi** [comment]
Actually, this is not perfect, because sometimes gives very slightly different (<1e-15) results for the same process.

**Comment by giampaolo** [comment]
Hi and sorry for the big delay in responding. What does this add? Nano seconds precision?

**Comment by gaborcsardi** [comment]
Yes, so it seems at https://github.com/giampaolo/psutil/blob/8b465260638cde5370808e23eb9ace608141393a/psutil/_psutil_windows.c#L589 that process creation time has a resolution of 1s on Windows. Which is OK, I guess, but we could do better, probably.

Instead of 
```c
    unix_time = ((LONGLONG)ftCreate.dwHighDateTime) << 32;
    unix_time += ftCreate.dwLowDateTime - 116444736000000000LL;
    unix_time /= 10000000;
    return Py_BuildValue("d", (double)unix_time);
```
we could simply do 
```c
  long long ll, secs, msecs;
  ll = ((LONGLONG) ft.dwHighDateTime) << 32;
  ll += ft.dwLowDateTime - 116444736000000000LL;
  secs = ll / 10000000;
  msecs = ll % 10000000;
  return Py_BuildValue("d", (double) secs + msecs / 10000000.0);
}
```

So this gives you microseconds.

One catch here is that on 32 bit processors, a 80 bit numerical register will be used for the double precision calculations, and this might cause non-deterministic results, depending on whether it is written to another registers or to memory or not. (The same problem exists on other 32 bit OSes as well AFAIK.)

To fix this you would either need to compare create_time with a tolerance, or store the original FILETIME in the process object, and use that for comparison.

**Comment by giampaolo** [comment]
> So this gives you microseconds.

OK. I'm wondering if there are circumstances where this matters. 

> Actually, this is not perfect, because sometimes gives very slightly different (<1e-15) results for the same process.

That is not good. Does your change introduces this behavior? Or it's already there?
There's some magic around process creation time because it is used to identify a process uniquely (see [here](https://github.com/giampaolo/psutil/blob/8b465260638cde5370808e23eb9ace608141393a/psutil/__init__.py#L381-L385) and [here](https://github.com/giampaolo/psutil/blob/8b465260638cde5370808e23eb9ace608141393a/psutil/__init__.py#L420-L423)). Different creation times for the same process would result in this:

```python
>>> from psutil import Process
>>> import os
>>> me = os.getdpid()
>>> Process(me) == Process(me)
False
>>> Process(me).is_running()
False
>>> Process(me).nice(3)
Error - Traceback exception: psutil.NoSuchProcess no process found with pid ...
```

That's because:
* `==` / `!=` operators are [overwritten](https://github.com/giampaolo/psutil/blob/8b465260638cde5370808e23eb9ace608141393a/psutil/__init__.py#L410-L418)
* [is_running()](https://github.com/giampaolo/psutil/blob/8b465260638cde5370808e23eb9ace608141393a/psutil/__init__.py#L562-L567) is used to make sure PID has not been [reused by another process](https://github.com/giampaolo/psutil/blob/8b465260638cde5370808e23eb9ace608141393a/psutil/__init__.py#L276-L285)
* process methods depending on it (like `terminate()`, `children()`) end up raising `NoSuchProcess` for a process which is alive
